opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -187,17 +147,116 @@ where
 #[derive(Clone)]
 pub struct PeriodicReader {
     exporter: Arc<dyn PushMetricsExporter>,
-    inner: Arc<Mutex<PeriodicReaderInner>>,
```

Comment:
Mutex requirement was removed and this eliminates a lot of potential deadlocks.

Reply:
Just to double check this is not related to async -> background change right? Just an addon optimzation?

Reply:
it is a required fix this to work, else flush/shutdown cannot operate without deadlocking. But I can try to make that change to own PR.

Reply:
It is infeasible to make this fix separate.